src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -258,17 +288,80 @@ private void EnsureGlobbingUrlBuilder()
             }
         }
 
-        private static void BuildLinkTag(IDictionary<string, string> attributes, TagHelperContent builder)
+        private void BuildLinkTag(IDictionary<string, string> attributes, TagHelperContent builder)
         {
             builder.Append("<link ");
 
             foreach (var attribute in attributes)
             {
-                builder.Append(
-                    string.Format(CultureInfo.InvariantCulture, "{0}=\"{1}\" ", attribute.Key, attribute.Value));
+                if (attribute.Key == "href")
+                {
+                    AppendHref(builder, attribute.Key, attribute.Value);
+                }
+                else
+                {
+                    builder.Append(
+                        string.Format(CultureInfo.InvariantCulture, "{0}=\"{1}\" ", attribute.Key, attribute.Value));
+                }
             }
 
             builder.Append("/>");
         }
+
+        private void AppendHref(TagHelperContent content, string hrefKey, string hrefValue)
+        {
+            string formatter = "{0}=\"{1}";
+            string fileVersion = string.Empty;
+            if (FileVersion == "true")
```

Comment:
What you're most likely seeing is due to pre-compilation essentially caching your page (assuming you're using pre-compilation, it's on by default in TagHelperSample.Web), just tried this on my box and it worked as expected but I ran into a pre-compilation caching issue because of it. 

/cc @DamianEdwards